etcdserver/api/v3rpc/util.go

```diff
@@ -49,7 +49,7 @@ func togRPCError(err error) error {
 	case etcdserver.ErrTooManyRequests:
 		return rpctypes.ErrTooManyRequests
 
-	case etcdserver.ErrNoLeader:
+	case etcdserver.ErrNoLeader, rpctypes.ErrNoLeader: // ErrNoLeader from leasehttp
```

Comment:
this is really goofy; maybe `lessor.LeaseTimeToLive` should still return io.EOF and v3api.LeaseTimeToLive will translate it to `etcdserver.ErrNoLeader`?